blog-server-services/src/impls/rbatis_post_service.rs

```diff
@@ -94,14 +121,64 @@ impl PostService for RbatisPostService {
         Ok(Post::count(&self.rb).await?)
     }
     async fn posts(&self, offset: &i64, limit: &i64) -> DResult<Vec<Post>> {
-        Ok(Post::select_all_with_limit_and_offset(&self.rb, limit, offset).await?)
+        let mut posts = Post::select_posts(&self.rb, limit, offset).await?;
+
+        let post_ids = posts.iter().map(|post| post.id).collect();
+        let grouped_tags: HashMap<i64, Vec<Tag>> = Post::select_tags_by_posts(&self.rb, post_ids)
+            .await?
+            .into_iter()
+            .fold(HashMap::new(), |mut map, dto| {
+                let key = dto.post_id;
+                let tag = dto.into();
+                map.entry(key).or_insert_with(Vec::new).push(tag);
+                map
+            });
+
+        for post in posts.iter_mut() {
+            match grouped_tags.get(&post.id) {
```

Comment:
We can replace `grouped_tags.get(&post.id)` to `grouped_tags.remove(&post.id)` to get owned object, and then we can not clone vector

Reply:
done....i miss NULL